src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs

```diff
@@ -458,15 +458,28 @@ public void RecordProperties(params string[] propertyNames)
 
         public Dictionary<string, string> GetPropertyValues(string testRoot, string configuration = "Debug", string targetFramework = null, string runtimeIdentifier = null)
         {
-            var propertyValues = new Dictionary<string, string>();
-
-            string intermediateOutputPath = Path.Combine(testRoot, Name, "obj", configuration, targetFramework ?? TargetFrameworks);
+            string finalOutputPath = Path.Combine(testRoot, Name, "obj", configuration, targetFramework ?? TargetFrameworks);
             if (!string.IsNullOrEmpty(runtimeIdentifier))
             {
-                intermediateOutputPath = Path.Combine(intermediateOutputPath, runtimeIdentifier);
+                finalOutputPath = Path.Combine(finalOutputPath, runtimeIdentifier);
             }
 
-            foreach (var line in File.ReadAllLines(Path.Combine(intermediateOutputPath, "PropertyValues.txt")))
+            return GetPropertyValues(finalOutputPath);
+        }
+
+        /// <summary>
+        /// Returns a dictionary of property name to value mappings.
+        /// </summary>
+        /// <param name="finalOutputPath">
+        /// The final output path after executing a dotnet command where propertyvalues.txt will be generated.
+        /// If the output path does not need to be hardcoded, use the overload requesting testRoot instead.
+        /// </param>
+        /// <returns></returns>
+        public Dictionary<string, string> GetPropertyValues(string finalOutputPath)
```

Comment:
The alternative I was thinking of was to not write the file in a RID-specific path.  IE update the injected target in `TestProject.Create` to something like this:

```c#
                string injectTargetContents =
    $@"<Project>
  <Target Name=`WritePropertyValues` BeforeTargets=`AfterBuild`>
    <ItemGroup>
{propertiesElements}
    </ItemGroup>
    <WriteLinesToFile
      File=`$(BaseIntermediateOutputPath)\$(Configuration)\$(TargetFramework)\PropertyValues.txt`
      Lines=`@(LinesToWrite)`
      Overwrite=`true`
      Encoding=`Unicode`
      />
  </Target>
</Project>";
```

Reply:
That makes sense. The `PublishRuntimeIdentifier` tests need this too and I added it there. I think I should retarget both of these to 7.0.2xx. And then merge the publish RID PR. Then merge this with 7.0.2xx and change the test.